### PR TITLE
Add metadata info

### PR DESCRIPTION
**Pull request by smercier**
please review this WMS Metadata enhancement

### CODE REVIEW

**Comment by LarsSchy** [comment]
The changes looks good to me !

File: chart-installation/generate_map_files/generate_map_config.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ def config_relative(path, args):
         chartsymbols = config_relative(chartsymbols, args)
 
     if not os.path.isfile(chartsymbols):
-        print("chartsymbols.xml not found at: " + chartsymbols)
+        print("chartsymbols.xml not found at: " + chartsymbols)
```
Comment (alexbrault) [inline chart-installation/generate_map_files/generate_map_config.py]:
```suggestion
        print("chartsymbols.xml not found at: " + chartsymbols)
```

File: chart-installation/generate_map_files/generate_map_config.py [hunk 2]
```diff
@@ -156,5 +169,7 @@ def config_relative(path, args):
               file=sys.stderr)
         sys.exit(1)
 
+    # TODO we should manage creating or not symbolset.
+    print("Generating symbolset ...")
     for symbolset in symbolsets:
-        generate_symbolset(symbolset, os.path.join(map_path, 'symbols'), False)
+        generate_symbolset(symbolset, os.path.join(map_path, 'symbols'), False,chartsymbols)
```
Comment (alexbrault) [inline chart-installation/generate_map_files/generate_map_config.py]:
```suggestion
        generate_symbolset(symbolset, os.path.join(map_path, 'symbols'), False, chartsymbols)
```

File: chart-installation/generate_map_files/mapgen/basechart.py [hunk 3]
```diff
@@ -171,6 +171,7 @@ def process_all_layers(data, target, config, point_table='Simplified',
             '3D Point': 'POINT',
         }
         print("Check geometry of all layers...")
+        #if 1==0:
```
Comment (alexbrault) [inline chart-installation/generate_map_files/mapgen/basechart.py]:
```suggestion
```

File: chart-installation/generate_map_files/mapgen/basechart.py [hunk 4]
```diff
@@ -310,22 +347,24 @@ def process_layer_colors(layer, color_table, input_file, msd, data, target,
...
+                    # we will push a readable name in metadata for this layer
+                    metadata_name = get_metadata_name(feature)
+                    # we will push a readdable Group name based on Navigation level
+                    group_layer = get_navigation_level(layer)
                     if geom == 'POINT':
                         layer_obj = chartsymbols.get_point_mapfile(
-                            layer, feature, 'default', msd,
-                            shp_fields[filename])
+                            layer, feature, group_layer, msd,
+                            shp_fields[filename],metadata_name)
```
Comment (alexbrault) [inline chart-installation/generate_map_files/mapgen/basechart.py]:
```suggestion
                            shp_fields[filename], metadata_name)
```

File: chart-installation/generate_map_files/mapgen/basechart.py [hunk 5]
```diff
@@ -310,22 +347,24 @@ def process_layer_colors(layer, color_table, input_file, msd, data, target,
...
                     elif geom == 'LINESTRING':
                         layer_obj = chartsymbols.get_line_mapfile(
-                            layer, feature, 'default', msd,
-                            shp_fields[filename])
+                            layer, feature, group_layer, msd,
+                            shp_fields[filename],metadata_name)
```
Comment (alexbrault) [inline chart-installation/generate_map_files/mapgen/basechart.py]:
```suggestion
                            shp_fields[filename], metadata_name)
```

File: chart-installation/generate_map_files/mapgen/basechart.py [hunk 6]
```diff
@@ -310,22 +347,24 @@ def process_layer_colors(layer, color_table, input_file, msd, data, target,
...
                     elif geom == 'POLYGON':
                         layer_obj = chartsymbols.get_poly_mapfile(
-                            layer, feature, 'default', msd,
-                            shp_fields[filename])
+                            layer, feature, group_layer, msd,
+                            shp_fields[filename],metadata_name)
```
Comment (alexbrault) [inline chart-installation/generate_map_files/mapgen/basechart.py]:
```suggestion
                            shp_fields[filename], metadata_name)
```

File: chart-installation/generate_map_files/mapgen/chartsymbols.py [hunk 7]
```diff
@@ -191,27 +191,26 @@ def load_lookups(self, root, point_style, area_style,
                 @ lookups_from_cs('SOUNDG', 'Point', 'X-SNDG')
             )
 
-    def get_point_mapfile(self, layer, feature, group, msd, fields):
+    def get_point_mapfile(self, layer, feature, group, msd, fields,metadata_name):
```
Comment (alexbrault) [inline chart-installation/generate_map_files/mapgen/chartsymbols.py]:
```suggestion
    def get_point_mapfile(self, layer, feature, group, msd, fields, metadata_name):
```

File: chart-installation/generate_map_files/mapgen/chartsymbols.py [hunk 8]
```diff
@@ -191,27 +191,26 @@ def load_lookups(self, root, point_style, area_style,
...
+                      fields, self.point_lookups.get(feature, []), self,
+                      metadata_name)
         if feature == 'LIGHTS':
             layer = LightsLayer(layer)
 
         return layer
 
     def get_line_mapfile(self, layer, feature_type, group, max_scale_denom,
-                         fields):
+                         fields,metadata_name):
```
Comment (alexbrault) [inline chart-installation/generate_map_files/mapgen/chartsymbols.py]:
```suggestion
                         fields, metadata_name):
```

File: chart-installation/generate_map_files/mapgen/chartsymbols.py [hunk 9]
```diff
@@ -191,27 +191,26 @@ def load_lookups(self, root, point_style, area_style,
...
         return Layer(layer, feature_type, 'LINE', group, max_scale_denom,
-                     fields, self.line_lookups.get(feature_type, []), self
-                     )
+                     fields, self.line_lookups.get(feature_type, []), self,
+                     metadata_name)
 
     def get_poly_mapfile(self, layer, feature_type, group, max_scale_denom,
-                         fields):
+                         fields,metadata_name):
```
Comment (alexbrault) [inline chart-installation/generate_map_files/mapgen/chartsymbols.py]:
```suggestion
                         fields, metadata_name):
```

File: chart-installation/generate_map_files/mapgen/generate_symbolset.py [hunk 10]
```diff
@@ -24,7 +24,7 @@
 symbolsets = ('day', 'dusk', 'dark')
 
 
-def generate_symbolset(symboltype, output_directory, force_update):
+def generate_symbolset(symboltype, output_directory, force_update,OCPN_lookuptable):
```
Comment (alexbrault) [inline chart-installation/generate_map_files/mapgen/generate_symbolset.py]:
```suggestion
def generate_symbolset(symboltype, output_directory, force_update, OCPN_lookuptable):
```

File: chart-installation/generate_map_files/mapgen/layer.py [hunk 11]
```diff
@@ -153,12 +154,26 @@ def __init__(self, layer_parent, geom_type, classes):
         self.geom_type = geom_type
         self.classes = classes
 
+    def get_human_readable_geomtype(self):
+        t = str(self.geom_type)
```
Comment (alexbrault) [inline chart-installation/generate_map_files/mapgen/layer.py]: You should just compare to the enum values; that will let the interpreter catch typos

Comment (smercier) [reply chart-installation/generate_map_files/mapgen/layer.py]: the proposed change is now applied in branch

**Pull request merged**